server/src/chatroomUsers.ts

```diff
@@ -0,0 +1,118 @@
+import express, { query } from "express";
+import { IChatRoom, IPinnedItem, error } from "./types";
+import db from "./firebase";
+import { IBackgroundState } from "./types";
+
+const collection = db.collection("chatroomUsers");
+
+const chatroomUserRouter = express.Router();
+
+//get all chatroom users
+chatroomUserRouter.get("/get-all", async (req, res) => {
+    const doc = await collection.get();
+
+    res.status(200).send(doc.docs.map(doc => doc.data()));
+})
+
+//create new chatroom user
+chatroomUserRouter.post("/add-user", async (req, res) => {
+    const { userId, screenName, avatar } = req.body as { userId: string, screenName: string, avatar: string };
+
+    const doc = await collection.doc(userId).get();
+
+    if (doc.exists) {
+        return error(res, "user already exists");
+    }
+
+    const newDoc = await collection
+        .doc(userId)
+        .set({ screenName, avatar });
+
+    res.status(200).send(newDoc);
+
+})
+
+//get chatroom user
+chatroomUserRouter.get("/get/:userId", async (req, res) => {
+    const { userId } = req.params as { userId: string};
+
+    const doc = await collection.doc(userId).get();
+
+    if(!doc.exists){
+        res.status(400).send({msg: "user does not exist"});
```

Comment:
I think this should return null instead of erroring, to allow the client to handle actual errors